### PR TITLE
fix(ESSNTL-4077): migrate sp-validator and rebuild-events-topic to confluent-kafka

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [ESSNTL-4077](https://issues.redhat.com/browse/ESSNTL-4077).
It fixes the sp-validator and rebuild-events-topic jobs so that they work with the confluent-kafka library.
I've also updated the test utils to reflect the structure and functionality of the confluent-kafka consumer object. I don't know how to fully test this without it integrating with the actual brokers, though, so this may require a follow-up.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by CodeHeeler — DISMISSED** [review]
LGTM

File: rebuild_events_topic.py [hunk 1]
```diff
@@ -42,38 +42,38 @@ def run(config, logger, session, consumer, event_producer, shutdown_handler):
...
-                    # metrics.ingress_message_handler_failure.inc()
-                    logger.exception("Unable to process message", extra={"incoming_message": message.value})
-
-            num_messages += len(messages)
-
+        new_messages = consumer.consume(timeout=10)
+        for message in new_messages:
+            try:
+                sync_event_message(json.loads(message.value()), session, event_producer)
+                # TODO: Metrics
```
Comment (thearifismail) [inline rebuild_events_topic.py:63]: Should these TODOs stay or go?

**Review by thearifismail — COMMENTED** [review]
The code looks good but all tests in test_event_rebuilder.py fail.

**Review by thearifismail — APPROVED** [review]
the problem with tests/test_event_rebuilder.py has been fixed and all 973 tests in the project passed.

File: rebuild_events_topic.py (hunk 1 above)
Comment (kruai) [reply rebuild_events_topic.py:63]: As far as I know, we don't really care about the metrics for these manual jobs, so I guess we can delete them?

**Pull request merged**